### PR TITLE
Sambanova models support

### PR DESCRIPTION
**Pull request by jhpiedrahitao**
PR implementing SambaNovaModel and allowing access to SambaNovaCloudModels.
includes, model, base tests and docs

### CODE REVIEW

**Comment by samuelcolvin** [comment]
Hi @jhpiedrahitao, I've never heard of Sambanova. Please can you start by creating an issue and see if we get significant interest from the community, before we consider adding this to `pydantic-ai` itself.

Alternatively if you want to use or even distribute this without waiting, you can always create a new package, e.g. `pyantic-ai-sambanova` which has `pydantic-ai` as a dependency, and exposes `SambaNovaModel`.

**Comment by sambarnes** [comment]
> Hi @jhpiedrahitao, I've never heard of Sambanova. Please can you start by creating an issue and see if we get significant interest from the community, before we consider adding this to `pydantic-ai` itself.

we've got a number of sambanova provided models at openrouter, and there is support for that already in pydantic-ai

i actually just saw this recent issue, after hearing users ask if it works with our services:
https://github.com/pydantic/pydantic-ai/issues/139

would you be open to a quick docs pr adding [another example at the end of here](https://ai.pydantic.dev/install/#custom-openai-client)?

```python
from openai import AsyncOpenAI

from pydantic_ai import Agent
from pydantic_ai.models.openai import OpenAIModel

client = AsyncOpenAI(
  base_url='https://openrouter.ai/api/v1',
  api_key='your-api-key',
)
model = OpenAIModel('anthropic/claude-3.5-sonnet', openai_client=client)
agent = Agent(model)
...
```

File: docs/install.md [hunk 1]
```diff
@@ -329,3 +329,50 @@ agent = Agent(model)
 To use [Ollama](https://ollama.com/), you must first download the Ollama client, and then download a model.
 
 You must also ensure the Ollama server is running when trying to make requests to it. For more information, please see the [Ollama documentation](https://github.com/ollama/ollama/tree/main/docs)
+
+### SambaNova
+
+To use [SambaNovaCloud](https://cloud.sambanova.ai/) through their API, go to [cloud.sambanova.ai/apis](https://cloud.sambanova.ai/apis) and click on Generate New API key.
```
[the remark this answers is not in the document]
Comment (snova-jorgep) [reply docs/install.md:335]: not modified, part of SambaNovaCloud urls

**Comment by jhpiedrahitao** [comment]
> Hi @jhpiedrahitao, I've never heard of Sambanova. Please can you start by creating an issue and see if we get significant interest from the community, before we consider adding this to `pydantic-ai` itself.
> 
> Alternatively if you want to use or even distribute this without waiting, you can always create a new package, e.g. `pyantic-ai-sambanova` which has `pydantic-ai` as a dependency, and exposes `SambaNovaModel`.

Hi @samuelcolvin I created the issue here https://github.com/pydantic/pydantic-ai/issues/218

You can find more information here [https://sambanova.ai/](https://sambanova.ai/) 
also in the issue linked to artificial  analysis providers leader board, where you can see SambaNova being on top for several models

File: docs/install.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhpiedrahitao) [reply docs/install.md:389]: not modified, part of SambaNovaCloud urls

File: docs/install.md [hunk 2]
```diff
@@ -383,3 +383,50 @@ agent = Agent(model)
 To use [Ollama](https://ollama.com/), you must first download the Ollama client, and then download a model.
 
 You must also ensure the Ollama server is running when trying to make requests to it. For more information, please see the [Ollama documentation](https://github.com/ollama/ollama/tree/main/docs)
+
+### SambaNova
```
Comment (soufianechami) [inline docs/install.md:387]: You may also need to add Sambanova LLM API to the list at the top of this document: 




> This installs the `pydantic_ai` package, core dependencies, and libraries required to use the following LLM APIs:
* [OpenAI API](https://platform.openai.com/docs/overview)
* [Google VertexAI API](https://cloud.google.com/vertex-ai) for Gemini models
* [Anthropic API](https://docs.anthropic.com/en/api/getting-started)
* [Groq API](https://console.groq.com/docs/overview)
* [SambaNova API](https://cloud.sambanova.ai/apis)

**Comment by samuelcolvin** [comment]
Closing as per https://github.com/pydantic/pydantic-ai/issues/218#issuecomment-2543169564

**Pull request closed without merging**